payments/models.py

```diff
@@ -135,6 +135,71 @@ def get_success_url(self) -> str:
     def get_process_url(self) -> str:
         return reverse('process_payment', kwargs={'token': self.token})
 
+    def get_payment_url(self):
```

Comment:
You can raise `RedirectNeeded` in you provider code. Any application using this lib should catch these and redirect as needed.

Is there a reason that won't work for you?